### PR TITLE
Contracts v4 changes

### PR DESCRIPTION
**Pull request by belyakove**

### CODE REVIEW

File: ZKSyncSDK/Classes/Model/TimeRange.swift [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+//  Created by Eugene Belyakov on 20/03/2021.
+//
+
+import Foundation
+
+public struct TimeRange: Decodable {
+    public var validFrom: UInt64
+    public var validUntil: UInt64
+    
+    public init(validFrom: UInt64, validUntil: UInt64) {
```
Comment (StanislavBreadless) [inline ZKSyncSDK/Classes/Model/TimeRange.swift:14]: There should be the ability to set the maximum possible time range by default

File: ZKSyncSDK/Classes/Model/Transactions/ChangePubKey.swift [hunk 2]
```diff
@@ -18,18 +18,21 @@ public class ChangePubKey: ZkSyncTransaction {
...
+    let timeRange: TimeRange
+    
     var signature: Signature?
-    var ethSignature: String?
+    var ethAuthData: T?
     
     var feeInteger: BigUInt { BigUInt(fee)! }
     
-    public init(accountId: UInt32, account: String, newPkHash: String, feeToken: UInt16, fee: String, nonce: UInt32) {
+    public init(accountId: UInt32, account: String, newPkHash: String, feeToken: UInt16, fee: String, nonce: UInt32, timeRange: TimeRange) {
```
Comment (StanislavBreadless) [inline ZKSyncSDK/Classes/Model/Transactions/ChangePubKey.swift:28]: The same goes here. The 99% of the users will not need to set timestamps on the transactions. They should be able to forget about them and not supply them when they don't need it

**Review by StanislavBreadless — COMMENTED** [review]
Generally great. There are a few points to consider.

Also, please make sure that the CI passes

File: ZKSyncSDK/Classes/Model/TimeRange.swift (hunk 1 above)
Comment (belyakove) [reply ZKSyncSDK/Classes/Model/TimeRange.swift:14]: Done

File: ZKSyncSDK/Classes/Model/Transactions/ChangePubKey.swift (hunk 2 above)
Comment (belyakove) [reply ZKSyncSDK/Classes/Model/Transactions/ChangePubKey.swift:28]: Done. For all different types of transactions.

**Pull request merged**